newsroom/template_filters.py

```diff
@@ -61,6 +70,110 @@ def parse_date(datetime):
+
+    if all_day:
+        return ScheduleType.ALL_DAY if duration in (0, DAY_IN_MINUTES) else ScheduleType.MULTI_DAY
+
+    if no_end_time:
+        return ScheduleType.NO_DURATION
+
+    if duration >= DAY_IN_MINUTES and start.date() != (end.date() if end else None):
+        return ScheduleType.MULTI_DAY
+
+    if duration <= DAY_IN_MINUTES and start.date() == (end.date() if end else None):
+        return ScheduleType.ALL_DAY
+
+    if duration == 0 and start.time() == (end.time() if end else None):
+        return ScheduleType.NO_DURATION
+
+    return ScheduleType.REGULAR
+
+
+def is_item_tbc(item: dict) -> bool:
+    event_tbc = item.get("event", {}).get("_time_to_be_confirmed", False)
+    planning = item.get("planning_items", [])
+    return event_tbc or (planning and planning[0].get("_time_to_be_confirmed", False))
+
+
+def format_event_datetime(item: dict) -> str:
+    date_info = item.get("dates", {})
+
+    if not date_info:
+        return ""
+
+    tz = date_info.get("tz", get_session_timezone())
+    try:
+        # Set the session timezone
+        set_session_timezone(tz)
+
+        start = parse_date(date_info.get("start"))
+        end = parse_date(date_info.get("end")) if date_info.get("end") else None
+        all_day = date_info.get("all_day")
+        no_end_time = date_info.get("no_end_time")
+        is_tbc_item = is_item_tbc(item)
+
+        schedule_type = get_schedule_type(start, end, all_day, no_end_time)
+
+        formatted_start = datetime_long(start)
+        formatted_end = datetime_long(end) if end else None
+
+        if is_tbc_item:
+            if schedule_type == ScheduleType.MULTI_DAY:
+                return lazy_gettext("Event Starts: {start} - Event Ends: {end} ({tz}) (Time to be confirmed)").format(
+                    start=formatted_start, end=formatted_end, tz=tz
+                )
+
+            else:
+                return lazy_gettext("Event Starts: {start} ({tz}) (Time to be confirmed)").format(
+                    start=formatted_start, tz=tz
+                )
+
+        elif schedule_type in (ScheduleType.ALL_DAY, ScheduleType.NO_DURATION):
+            return lazy_gettext("Event Starts: {start} ({tz})").format(start=formatted_start, tz=tz)
+
+        elif schedule_type == ScheduleType.MULTI_DAY:
+            return lazy_gettext("Event Starts: {start} - Event Ends: {end} ({tz})").format(
+                start=formatted_start, end=formatted_end, tz=tz
+            )
+        elif schedule_type == ScheduleType.REGULAR:
+            return lazy_gettext("Event Starts: {start_time} to {end_time} - On Date: {date} ({tz})").format(
+                start_time=notification_time(start),
+                end_time=notification_time(end),
+                date=notification_date(start),
+                tz=tz,
+            )
+        else:
+            # Default
+            return lazy_gettext(
+                "Event Starts: {start_time} {start_date} to Event Ends: {end_time} {end_date} ({tz})"
```

Comment:
I'm not sure about the `Event starts: ... to Event Ends: ...`, what about:
```
On: {start_time} {start_date} to {end_time} {end_date}
```
or maybe using `Date:` as a prefix? what do you think @MarkLark86

Reply:
makes sense

Reply:
Yeah I agree @petrjasek, maybe just `On: ` makes more sense as it may not always be an Event right!

Reply:
should we only change the default case, or should we change them all?

Reply:
all of them